text/0055-remove-unspecified.md

```diff
@@ -0,0 +1,194 @@
+The approach above can be extended to this case too: make the action apply to an unspecified principal _and_ resource.
+But now "unspecified" entities are being used to represent multiple distinct concepts: a file store, (unidentified) users creating an account, and a system that manages accounts.
+
+This RFC proposes that users should instead explicitly create these "dummy" entities in an application-specific manner.
+For this example, the user should create entities of three types: `FileSystem`, `UnauthenticatedUser`, and `AccountManager`.
+The first will act as the principal for a "getFile" request, and the second and third will act as the principal and resource for a "createAccount" request.
+
+This is in line with [our previous suggestion](https://cedar-policy.slack.com/archives/C0547KH7R19/p1706656288284189) to use an application-specific `Unauthenticated` type to represent "unauthenticated" users, rather than using unspecified entities.
+
+## Motivation
+
+Unspecified entities are a confusing concept, as evidenced by messy special casing in our own code and multiple RFCs to refine the behavior ([RFC35](https://github.com/cedar-policy/rfcs/pull/35), [RFC47](https://github.com/cedar-policy/rfcs/pull/47)).
+We expect that this concept will only become more confusing once we stabilize the partial evaluation experimental feature, which allows “unknown” entities that are subtly different from “unspecified” entities.
+Unspecified and unknown entities are both pre-defined "dummy" entities that act like any other entity, as long as they are unused or only used in a trivial way (e.g., `principal == principal`).
+If a policy uses an _unknown entity_ in a non-trivial way, then evaluation will produce a residual.
+If a policy uses an _unspecified entity_ in a non-trivial way, then evaluation will produce an error.
+
+Unspecified entities also lead to two sharp corners in our public API:
+
+1. Omitting a `appliesTo` field in the schema (or setting it to null) means that an action applies only to unspecified entities, while using an empty list means that it applies to no entities. There is no way to say that an action applies to both unspecified entities and other types of entities. For more discussion, see [RFC35](https://github.com/cedar-policy/rfcs/pull/35). (Note: RFC35 will be closed in favor of the current RFC, if accepted.)
+2. It is unclear how we should create `Request`s that use both unspecified and unknown entities. This has not been an issue so far since partial evaluation and its APIs are experimental, but we are looking to stabilize them soon. Currently the best (although not ideal) proposal is to use a builder pattern for `Request`s (e.g., `request.principal(Some(User::"alice"))` sets the principal field of a request). To set a field to be “unspecified” users should pass in `None` instead of `Some(_)`; to set a field to be unknown users should not call the field constructor at all.
```

Comment:
Is this the most intuitive?  Have we considered whether doing it the other way around (None = unknown, not calling = unspecified) would be more intuitive for users?

Reply:
Fair question. I think if we keep our current `Request::new` api in addition to the builder pattern, then we need to stick with `None` == unspecified, since this is already the precedent. But if we get rid of the current constructor, then I'm open to swapping.

Reply:
> if we get rid of the current constructor, then I'm open to swapping.

Even we get rid of the current constructor and think swapping would be more intuitive from a blank slate, I still worry it would be confusing for people who already used `None` for unspecified.